FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -336,33 +336,50 @@ - (BOOL)isCodeHashIdentical:(NSString *)codeHash {
   return codeHash && [codeHash isEqualToString:[machO codeHash]];
 }
 
+- (NSString *)getCodeHash {
+  NSString *executablePath = [[NSBundle mainBundle] executablePath];
+  FIRAppDistributionMachO *machO = [[FIRAppDistributionMachO alloc] initWithPath:executablePath];
+  return [machO codeHash];
+}
+
 - (void)startFeedbackWithAdditionalFormText:(NSString *)additionalFormText {
-  UIImage *screenshot = [FIRFADInAppFeedback captureProgrammaticScreenshot];
-  if ([self isTesterSignedIn]) {
-    [self.uiService startFeedbackWithAdditionalFormText:additionalFormText image:screenshot];
-  } else {
-    [self signInTesterWithCompletion:^(NSError *_Nullable error) {
-      if (error) {
-        return;
-      }
-      [self.uiService startFeedbackWithAdditionalFormText:additionalFormText image:screenshot];
-    }];
-  }
+  [self startFeedbackWithAdditionalFormText:additionalFormText image:nil];
 }
 
-- (void)startFeedbackWithAdditionalFormText:(NSString *)additionalFormText image:(UIImage *)image {
+- (void)startFeedbackWithAdditionalFormText:(NSString *)additionalFormText
+                                      image:(UIImage *_Nullable)image {
   if ([self isTesterSignedIn]) {
-    [self.uiService startFeedbackWithAdditionalFormText:additionalFormText image:image];
+    [self findReleaseAndStartFeedbackWithAdditionalFormText:additionalFormText image:image];
   } else {
     [self signInTesterWithCompletion:^(NSError *_Nullable error) {
       if (error) {
         return;
       }
-      [self.uiService startFeedbackWithAdditionalFormText:additionalFormText image:image];
+      [self findReleaseAndStartFeedbackWithAdditionalFormText:additionalFormText image:image];
     }];
   }
 }
 
+- (void)findReleaseAndStartFeedbackWithAdditionalFormText:(NSString *)additionalFormText
+                                                    image:(UIImage *_Nullable)image {
+  [FIRFADApiServiceSwift
```

Comment:
TODO (for later): Because network requests can be slow, we should probably consider doing this during app start which checks whether the tester has access to this build?